hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/EditLogFileOutputStream.java

```diff
@@ -84,9 +84,9 @@ public EditLogFileOutputStream(Configuration conf, File name, int size)
     doubleBuf = new EditsDoubleBuffer(size);
     RandomAccessFile rp;
     if (shouldSyncWritesAndSkipFsync) {
```

Comment:
+1, but we should change it as following? it seems that it is inconsistent with parameter 'shouldSyncWritesAndSkipFsync'?
```
    if (shouldSyncWritesAndSkipFsync) {
      rp = new RandomAccessFile(name, "rw");
    } else {
      rp = new RandomAccessFile(name, "rwd");
    }
```

Reply:
`shouldSyncWritesAndSkipFsync=true` means that we will use `RandomAccessFile(name, "rwd")` to sync data instead of `fc.force(false)`.

`shouldSyncWritesAndSkipFsync=false` means that we will use `fc.force(false)` to sync data, here `rp = new RandomAccessFile(name, "rw");`

Reply:
Yes, absolutely right! It's very interesting journey while try to review this code segment again, I have checked in one unreasonable improvement (HDFS-15882) because I mislead this parameter meaning.
Now this changes is pretty good and is expected.
LGTM. +1.